### ISSUE TITLE
"Should Bitwarden remember..." bar is blinking

### ISSUE THREAD

**Issue by sitnarf** [bitwarden/browser#1243]
Using Ubuntu 20.04 and Chrome/Chromium, the bar is blinking (see the video). 
https://www.dropbox.com/s/loexvau590r7fvp/vokoscreen-2020-05-22_18-08-49.mkv?dl=0

**Comment by gitthangbaby** [comment]
i wish the bar showed up at all in Firefox :)

**Comment by jDally987** [comment]
Yeah I've had this too for a long time

Doesn't really bother me though

**Comment by threesixandzero** [comment]
it is a bit annoying, to be honest.. particularly when trying to click the "save" button and being unable to do so without a form of epilepsy. any chance a fix has been presented? happens when i run BW on chrome, win10pro environment.

**Comment by timkley** [comment]
Can confirm: happening on Mac OS 10.15.7 and Chrome 88, too.

**Comment by eliykat** [comment]
Can you please link me to a website you're experiencing this issue on?

**Comment by timkley** [comment]
It's happening on this site for me: https://demo.mauve.de/

If you click on the user icon and enter any credentials you'll be redirected and the flashing happens. Same if you visit https://demo.mauve.de/account/login directly and try to login with any credentials.

**Comment by turkeyphant** [comment]
Been happening for as long as I can rememeber on Windows and macOS as well I think. A variety of sites. Most recently giffgaff.com

**Comment by Asafkbalink** [comment]
I'm having the same, Ubuntu 20.04 & Chrome 88. Happens on my windows machine as well. Glad to see it being acknowledged!

**Comment by dave7280** [comment]
I see it has been known for almost 2 years. Are you **EVER** going to fix it?
Very bad first impression to users.

**Comment by linaspasv** [comment]
Any ETA when this will be fixed? It's very annoying, happens to most of the websites on Chrome.

**Comment by tesiek** [comment]
It also happend on Firefox (Windows 10). every site blinking when password remember prompt. Can live with that but it is annoying, that's for sure.

**Comment by fixator10** [comment]
`Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.192 Safari/537.36 OPR/74.0.3911.218`, same happens.

On some sites its flickers very fast, on some - not. It also seems like it switches between two variants of that bar, one with "Never for this website/Yes, save now" and another one with just "Never/Save"

**Comment by lundmikkel** [comment]
I'm having this issue on most websites (Windows 10/Chrome). It even shows the flashing banner for sites I've just logged into using Bitwarden!?

**Comment by mikarama** [comment]
Got it aswell on about everywebsite. Windows/Chrome last versions

**Comment by aliekiddo** [comment]
Happens on all sites Version 89.0.4389.114 (Official Build) (x86_64)

**Comment by toktor** [comment]
+1.......................................

**Comment by ikwtif** [comment]
also happens on Brave browser: Version 1.23.71 Chromium: 90.0.4430.72 (Official Build) (64-bit)
almost impossible to click on something

**Comment by kasumiru** [comment]
windows 10, chrome 90.0.4430.93, Bitwarden bar is ugly blinking.

**Comment by com-xuonghuynh** [comment]
They can't fix it, disappointed

**Comment by timkley** [comment]
@eliykat can we provide more information for how to reproduce this issue?

**Comment by aliekiddo** [comment]
@timkley Just log into a website where you don't have an account stored in Bitwarden. That's all you have to do.

**Comment by eliykat** [comment]
I've been unable to reproduce this, but clearly enough people are experiencing it.

@timkley (or anyone else) the following information would be really helpful:
* does it make any difference whether you have login(s) saved for the site?
* does it make any difference whether your window is maximized or windowed?
* do you have any error messages in your console while the bar is flickering? If so can you copy and paste or screenshot those errors (right-click the page, click "inspect", then click "console")

**Comment by turkeyphant** [comment]
> I've been unable to reproduce this, but clearly enough people are experiencing it.
> 
> @timkley (or anyone else), I have a few theories, but the following information would be really helpful to narrow it down:
> 
> * does it make any difference whether you have login(s) saved for the site?

I think it only happens with logins saved for me

> * does it make any difference whether your window is maximized or windowed?

No

**Comment by aliekiddo** [comment]
> does it make any difference whether you have login(s) saved for the site?
No, It happens both when I do have a login stored in BW but I'm not logging into that account - i.e. I log into my work's PayPal stored in my work's 1Password (I have my personal PayPal account stored in BW, but not my work account) and when I log into a website where I don't have an account already stored with BW - i.e. I'm creating a new account for a site or logging into a work account with 1Password and I don't have an account for that site stored in BW

> does it make any difference whether your window is maximized or windowed?
No difference

> do you have any error messages in your console while the bar is flickering? If so can you copy and paste or screenshot those errors (right-click the page, click "inspect", then click "console")
No errors

> do you have any other extensions installed that could affect web page content? if so, does it make any difference whether you disable them?
1Password - but this also happens on FireFox where I don't have 1Password installed
LastPass (Inactive) - still happens when disabled
AdblockPlus - still happens when disabled

**Comment by timkley** [comment]
@eliykat 

I just created a fresh Chrome Profile to test this and couldn't reproduce the issue with only Bitwarden active. Maybe we can identify the root cause by collecting responses which extensions are installed? Maybe the logged in state of Chrome plays a role here?

I tried to debug the extension itself and now the problem is gone 🤔 

Could it be that by debugging I loaded new assets from the extension instead of old ones?

**Comment by aliekiddo** [comment]
I've just updated my Chrome and it's stopped flashing for now - It could have something to do with a pending update status - my theory for this is based on that the flashing changed momentarily as I hovered over the Update button next to my extensions row.

**Comment by HitoTech** [comment]
Happening to me too, on Brave, Linux AND MacOS.

I've been using Bitwarden since 3 months, and I always got this bug, on multiple Brave versions.

**Comment by eduardoarandah** [comment]
May I suggest moving this dialog on the icon?
Changing the DOM seems like the perfect ground for bugs.

**Comment by qwx** [comment]
Well, I try to debug this, and I found something wrong.
If there is another tab keep changing title in background, the banner keep blink,
the code
```
this.tabs.onUpdated.addListener(async (tabId: any, changeInfo: any, tab: any) => {
            if (this.main.onUpdatedRan) {
                return;
            }
            this.main.onUpdatedRan = true;
            await this.main.checkNotificationQueue();
            await this.main.refreshBadgeAndMenu();
            this.main.messagingService.send('tabUpdated');
            this.main.messagingService.send('tabChanged');
        });
```
in tabs.background.ts keep running so the addLogin message in queue will delete the bar then add it again.
but i don't know how to fix this, Please Help, it'a annoying.

**Comment by eliykat** [comment]
Hey @qwx that's a great find! I had looked into this, but I couldn't reliably reproduce it or spot where the likely error was. With this info, I'll schedule it to do a deep dive and hopefully come back with a fix.

@eduardoarandah I would also prefer this, but we can't open the popup programatically (see the discussion [here](https://github.com/bitwarden/browser/pull/987)). We're stuck in the DOM for the time being.

**Comment by igorsantos07** [comment]
I wonder if you could create a desktop notification instead of the bar?

**Comment by uiii** [comment]
I experienced the same issue. But all other password managers work without any issue in this, there must be some good way how to do the overlay properly.

**Comment by uiii** [comment]
I have limited knowledge how to debug the extensions, but this function https://github.com/bitwarden/browser/blob/d4231589ad10c1cf2567beb8c86f0ff4148147f0/src/content/notificationBar.ts#L48 is called repeatedly with 
```
{
    "command": "openNotificationBar",
    "data": {
        "type": "add"
    }
}
```

**Comment by uiii** [comment]
@eliykat If you can instruct me, I can provide you more info.

**Comment by luponata** [comment]
Why is this issue closed ? the problem is still present

Chrome 96.0.4664.45 on Ubuntu 20.04.3 LTS

**Comment by djsmith85** [comment]
@uiii You are correct that this method is called often, as the save prompt might need to be re-displayed if there is updates to the tab. It would be out of scope for this comment to explain all the inner workings of the save prompt. As a starting point for you, if you want to dive in further: I'd say have a look at the `tabs.background.ts` which is one area that calls checkNotificationQueue , `notification.background.ts` that has the notificationQueue, and adds messages to it when changes to a page with the target fields have been detected. 

@luponata This issue was closed as my work in #2129 was merged. Those changes were not released yet, but will be included in the next release, which will be very soon.

**Comment by uiii** [comment]
I can confirm this is probably related somehow to some other extensions, because it wasnt happening in a clear browser profile. 

I am thinking if you take a good approach how to do it if other password managers doesnt have the same problem.
